rfcs/models/security/confidentiality-design-choices.md

```diff
@@ -0,0 +1,174 @@
+<!--
+ Copyright IBM Corp. All Rights Reserved.
+
+ SPDX-License-Identifier: CC-BY-4.0
+ -->
+# Design Choices for End-to-End Confidentiality Mechanism
+
+- RFC: 01-008-appendix
+- Authors: Venkatraman Ramakrishna, Dhinakaran Vinayagamurthy, Sandeep Nishad, Krishnasuri Narayanam, Sikhar Patranabis
+- Status: Proposed
+- Since: 10-May-2022
+
+
+## Summary
+
+To fulfil the requirements described in the [end-to-end confidentiality spec](./confidentiality.md), the view contents returned by the source ledger must contain some unencrypted portions in the signed payload that can validate the plaintext contents as decrypted by the destination client. This allows the destination peers to validate the plaintext against tampering by the client. The solution for this was to add a hash of some form within the interoperation payload in the view.
+
+## General Considerations
+
+For private (confidential) portions of a view's contents:
+- They should be encrypted with the receiver's public key.
+- If they only need tamper protection, they should be hashed.
+- If they need tamper protection and need to be authenticated by the receiver, they should be hashed and signed.
+For public (non-confidential) portions of a view's contents:
+- If they need to be authenticated by the receiver, they should be signed.
+- If they only need tamper protection, they can be either hashed or signed.
+(_Note_: "hashed" implies "HMAC")
+
+## View Generation Options
+
+There are different mechanisms and protocols for hash creation and verification. We will list all of these below, with comments on their security and usability. The following key is used:
+- _DIM_: Destination Interoperation Module
+- _DP_: Destination Peer
+- _DC_: Destination Client
+- _SIM_: Source Interoperation Module
+- _SP_: Source Peer
+- _M_: Message, i.e., confidential view contents
+- _C_: Ciphertext, i.e., encrypted data
+- _P_: Payload, i.e., of the view
+- _Sigma_: Signature over view payload
+
+### Protocol #1
+
+Use SHA256 to hash the confidential view contents, and supply the hash as plaintext in the view.
+
+SIM computes:
+```
+C = Encrypt(M)
+P = C || SHA256Hash(M)
+```
+SP computes:
+```
+Sigma = Sign(P)
+```
+DC computes:
+```
+M' = Decrypt(C)
+```
+DIM computes:
```

Comment:
You might also want to say what all DIM gets from DC: M', P, Sigma.

Reply:
I don't think this is necessary. What the DIM gets is indicated by what it can process (in the subsequent lines). I want to keep this minimal.